runtime/jcl/common/shared.c

```diff
@@ -80,7 +80,8 @@ static jint createROMClassCookie(JNIEnv* env, J9JavaVM* vm, J9ROMClass* romClass
 static jint createToken(JNIEnv* env, jint helperID, J9ClassPathEntry** cpEntry_, const char* tokenChars, jsize tokenSize);
 static UDATA correctURLPath(JNIEnv* env, const char* pathChars, jsize pathLen, char** correctedPathPtr, J9SharedStringFarm** jclStringFarm);
 static const char* getCachedString(JNIEnv* env, const char* input, jsize length, J9SharedStringFarm** farmRoot, const J9UTF8** getUTF8);
-static UDATA getCpeTypeForProtocol(char* protocol, jsize protocolLen, const char* pathChars, jsize pathLen);
+static UDATA getCpeTypeForProtocol(JNIEnv *env, char *protocol, jsize protocolLen, const char *urlPathChars, jsize urlPathLen, char **correctedPathPtr);
```

Comment:
The second parameter should be `const char *protocol` - the function does not need write access to that string.